plugins/catalog-react/src/components/EntityRefLink/EntityRefLink.tsx

```diff
@@ -39,6 +61,104 @@ export type EntityRefLinkProps = {
   children?: React.ReactNode;
 } & Omit<LinkProps, 'to'>;
 
+type PeekAheadPopoverProps = {
+  popupState: PopupState;
+  entityRef: CompoundEntityRef;
+};
+
+const useStyles = makeStyles(() => {
+  return {
+    popoverPaper: {
+      width: '20em',
+    },
+  };
+});
+
+export const PeekAheadPopover = ({
+  popupState,
+  entityRef,
+}: PeekAheadPopoverProps) => {
+  const entityRoute = useRouteRef(entityRouteRef);
+  const classes = useStyles();
+  const catalogApi = useApi(catalogApiRef);
+
+  const {
+    value: entity,
+    loading,
+    error,
+  } = useAsync(async () => {
+    if (popupState.isOpen) {
+      const retrievedEntity = await catalogApi.getEntityByRef(entityRef);
+      if (!retrievedEntity) {
+        throw new Error(`${entityRef.name} was not found`);
+      }
+      return retrievedEntity;
+    }
+    return undefined;
+  }, [popupState]);
+
+  if (loading) {
+    return null;
```

Comment:
Would probably make sense to have a `<Progress />` or even a [mui Skeleton](https://v4.mui.com/components/skeleton/) here

Reply:
I was thinking about his. If you hover, you would see a loading thing. Would it be better to see the info we already have while it is loading? If we dont should anything except a loading bar, I think its going to be a bit strange for the user if the catalog is slow. In anycase, Ill try the options out.

Reply:
Does this look ok:

![image](https://user-images.githubusercontent.com/553697/206207729-c5cbae8a-ea54-4ec3-b97c-7a13905e2753.png)

Reply:
Then when it is done loading:

![image](https://user-images.githubusercontent.com/553697/206207843-e382d3d8-8412-40cb-817e-f2de0c2433e2.png)